Changelog.md

```diff
@@ -8,6 +8,9 @@
  * Add `IsList` instances
  * Deprecate `Data.ByteString.Lazy.Builder`
  * Add `partition` to `Data.ByteString.Char8` and `Data.ByteString.Lazy.Char8`
+ * Add `unsafePackLiteral` to `Data.ByteString.Internal`. Where possible,
+   use known-key variant of C `strlen` from `GHC.CString` that supports
+   constant folding.
```

Comment:
What does "known-key" mean here?

I think it would also be nice if the changelog entry was more explicit about the potential benefits for users.

Reply:
It's a term used in GHC that means that the compiler knows about the identifier and consequently can do sophisticated rewrites with it. Ascending levels of magic:

* Ordinary function (no magic)
* Known-key function (implementation is in library space in `base` but GHC's built-in rewrite rules can look for the function)
* Primop (implementation is provided by the compiler itself, and GHC's built-in rewrite rules can look for the function)

I can beef up the changelog entry.

Reply:
I've improved this by adding a link to the GHC wiki entry on known-key things and by clarifying that string literal desugar to ByteString via `OverloadingStrings` get better. Let me know if there's anything else.